services/server/src/server/services/storageServices/EtherscanVerifyAPIService.ts

```diff
@@ -0,0 +1,586 @@
+    } catch (error) {
+      logger.error("Failed to fetch Routescan chain list", {
+        workspace,
+        endpoint,
+        error,
+      });
+      throw new Error(
+        `Failed to fetch Routescan chain list for ${workspace} from ${endpoint}`,
+      );
+    }
+
+    if (!response.ok) {
+      const body = await response.text();
+      logger.error("Routescan chain list returned non-OK response", {
+        workspace,
+        endpoint,
+        status: response.status,
+        body,
+      });
+      throw new Error(
+        `Failed to load Routescan chain list for ${workspace} (${response.status}): ${body}`,
+      );
+    }
+
+    const payload = (await response.json()) as {
+      items?: RoutescanEntry[];
+    };
+
+    if (!Array.isArray(payload.items)) {
+      throw new Error(
+        `Invalid Routescan chain list payload for workspace ${workspace}`,
+      );
+    }
+
+    for (const entry of payload.items) {
+      if (entry == null) continue;
+
+      const { chainId, evmChainId } = entry;
+
+      if (
+        chainId === undefined ||
+        chainId === null ||
+        evmChainId === undefined ||
+        evmChainId === null
+      ) {
+        logger.warn("Skipping Routescan entry missing required fields", {
+          workspace,
+          entry,
+        });
+        continue;
+      }
+
+      try {
+        const url = new URL("https://api.routescan.io/");
+        url.pathname = `/v2/network/${encodeURIComponent(
+          workspace,
+        )}/evm/${encodeURIComponent(String(chainId))}/etherscan/api`;
+        url.search = "";
+        url.hash = "";
+
+        apiUrls[String(evmChainId)] = url.toString();
+      } catch (error) {
+        logger.warn("Skipping invalid Routescan API URL", {
+          workspace,
+          entry,
+          error,
+        });
+      }
+    }
+  }
+
+  if (Object.keys(apiUrls).length === 0) {
+    throw new Error(
+      "Routescan chain list did not contain any usable endpoints",
+    );
+  }
+
+  return apiUrls;
+};
+
+export const getEtherscanApiKeyEnvironmentVariables = (): Record<
+  string,
+  string
+> => {
+  const result: Record<string, string> = {};
+  for (const [chain, chainProperties] of Object.entries(sourcifyChainsMap)) {
+    if (
+      chainProperties.supported &&
+      chainProperties.etherscanApi &&
+      chainProperties.etherscanApi.apiKeyEnvName
+    ) {
+      result[chain] = chainProperties.etherscanApi.apiKeyEnvName;
+    }
+  }
+  return result;
+};
+
+interface EtherscanRpcResponse {
+  status: "0" | "1";
+  message: string;
+  result: string;
+}
+
+export interface EtherscanVerifyAPIServiceOptions {
+  /** Mapping of chainId to the explorer API base URL */
+  chainApiUrls?: ChainApiUrls;
+  /** Optional mapping of chainId to API keys */
+  apiKeys?: Record<string, string>;
+  /** Optional fallback API key when chain-specific key is missing */
+  defaultApiKey?: string;
+}
+
+export class EtherscanVerifyAPIService implements WStorageService {
```

Comment:
I think we should be consistent about the naming style.

```suggestion
export class EtherscanVerifyApiService implements WStorageService {
```

I know there are different approaches to acronyms for camel case. I personally prefer to treat acronyms like any other word. In any case, we should try to be consistent about this, in the same file at the very least (you are writing `getEtherscanApiKeyEnvironmentVariables` and `EtherscanRpcResponse`).